src/modals/WalletConnect/WalletConnectSessionProposalModal.tsx

```diff
@@ -0,0 +1,200 @@
+import { Trans, useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+
+import AssetLogo from '@/components/AssetLogo'
+import InfoBox from '@/components/InfoBox'
+import AddressSelect from '@/components/Inputs/AddressSelect'
+import { Section } from '@/components/PageComponents/PageContainers'
+import Paragraph from '@/components/Paragraph'
+import { useAppDispatch, useAppSelector } from '@/hooks/redux'
+import useAddressGeneration from '@/hooks/useAddressGeneration'
+import CenteredModal, { ModalFooterButton, ModalFooterButtons } from '@/modals/CenteredModal'
+import { selectAddressesInGroup } from '@/storage/addresses/addressesSelectors'
+import { saveNewAddresses } from '@/storage/addresses/addressesStorageUtils'
+import { networkPresetSwitched } from '@/storage/settings/networkActions'
+import { Address } from '@/types/addresses'
+import { SessionProposalEvent } from '@/types/walletConnect'
+import { getRandomLabelColor } from '@/utils/colors'
+import { cleanUrl } from '@/utils/misc'
+import { isNetworkValid, parseSessionProposalEvent } from '@/utils/walletConnect'
+
+interface WalletConnectSessionProposalModalProps {
+  approveProposal: (signerAddress: Address) => Promise<void>
+  rejectProposal: () => Promise<void>
+  proposalEvent: SessionProposalEvent
+  onClose: () => void
+}
+
+const WalletConnectSessionProposalModal = ({
+  onClose,
+  approveProposal,
+  rejectProposal,
+  proposalEvent
+}: WalletConnectSessionProposalModalProps) => {
+  const { t } = useTranslation()
+  const posthog = usePostHog()
+  const currentNetworkId = useAppSelector((s) => s.network.settings.networkId)
+  const currentNetworkName = useAppSelector((s) => s.network.name)
+  const dispatch = useAppDispatch()
+  const { requiredChainInfo, metadata } = parseSessionProposalEvent(proposalEvent)
+  const addressesInGroup = useAppSelector((s) => selectAddressesInGroup(s, requiredChainInfo?.addressGroup))
+  const { generateAddress } = useAddressGeneration()
+
+  const [signerAddress, setSignerAddress] = useState<Address>()
+
+  const group = requiredChainInfo?.addressGroup
+
+  const showNetworkWarning =
+    requiredChainInfo?.networkId && !isNetworkValid(requiredChainInfo.networkId, currentNetworkId)
+
+  useEffect(() => {
+    setSignerAddress(
+      addressesInGroup.length > 0 ? addressesInGroup.find((a) => a.isDefault) ?? addressesInGroup[0] : undefined
+    )
+  }, [addressesInGroup])
+
+  const handleSwitchNetworkPress = async () => {
+    if (requiredChainInfo?.networkId === 'mainnet' || requiredChainInfo?.networkId === 'testnet') {
+      dispatch(networkPresetSwitched(requiredChainInfo?.networkId))
+    }
+  }
+
+  const generateAddressInGroup = () => {
+    const address = generateAddress({ group })
+    saveNewAddresses([{ ...address, isDefault: false, color: getRandomLabelColor() }])
+
+    posthog.capture('New address created through WalletConnect modal')
+  }
+
+  const rejectAndCloseModal = async () => {
```

Comment:
This method is async but nothing inside is awaited. Should `rejectProposal` be awaited?

Reply:
I'll address this as a commit in the monorepo, if you don't mind!